eureka-client/src/main/java/com/netflix/discovery/DiscoveryClient.java

```diff
@@ -1380,60 +1406,65 @@ public HealthCheckHandler getHealthCheckHandler() {
                         }
-                    } else {
-                        // Just refresh mapping to reflect any DNS/Property change
-                        instanceRegionChecker.getAzToRegionMapper().refreshMapping();
                     }
+                } else {
+                    // Just refresh mapping to reflect any DNS/Property change
+                    instanceRegionChecker.getAzToRegionMapper().refreshMapping();
                 }
+            }
 
-                boolean success = fetchRegistry(remoteRegionsModified);
-                if (success) {
-                    registrySize = localRegionApps.get().size();
-                    lastSuccessfulRegistryFetchTimestamp = System.currentTimeMillis();
-                }
+            boolean success = fetchRegistry(remoteRegionsModified);
+            if (success) {
+                registrySize = localRegionApps.get().size();
+                lastSuccessfulRegistryFetchTimestamp = System.currentTimeMillis();
+            }
 
-                if (logger.isDebugEnabled()) {
-                    StringBuilder allAppsHashCodes = new StringBuilder();
-                    allAppsHashCodes.append("Local region apps hashcode: ");
-                    allAppsHashCodes.append(localRegionApps.get().getAppsHashCode());
-                    allAppsHashCodes.append(", is fetching remote regions? ");
-                    allAppsHashCodes.append(isFetchingRemoteRegionRegistries);
-                    for (Map.Entry<String, Applications> entry : remoteRegionVsApps.entrySet()) {
-                        allAppsHashCodes.append(", Remote region: ");
-                        allAppsHashCodes.append(entry.getKey());
-                        allAppsHashCodes.append(" , apps hashcode: ");
-                        allAppsHashCodes.append(entry.getValue().getAppsHashCode());
-                    }
-                    logger.debug("Completed cache refresh task for discovery. All Apps hash code is {} ",
-                            allAppsHashCodes.toString());
+            if (logger.isDebugEnabled()) {
+                StringBuilder allAppsHashCodes = new StringBuilder();
+                allAppsHashCodes.append("Local region apps hashcode: ");
+                allAppsHashCodes.append(localRegionApps.get().getAppsHashCode());
+                allAppsHashCodes.append(", is fetching remote regions? ");
+                allAppsHashCodes.append(isFetchingRemoteRegionRegistries);
+                for (Map.Entry<String, Applications> entry : remoteRegionVsApps.entrySet()) {
+                    allAppsHashCodes.append(", Remote region: ");
+                    allAppsHashCodes.append(entry.getKey());
+                    allAppsHashCodes.append(" , apps hashcode: ");
+                    allAppsHashCodes.append(entry.getValue().getAppsHashCode());
                 }
-            } catch (Throwable th) {
```

Comment:
As this is thread outermost loop try/catch we should catch Throwable exceptions. Otherwise the registry refresher may silently fail without us knowing about it.